File: csharp/calculator-conundrum/CalculatorConundrum.cs
using System;

public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string operation)
    {
        try
        {
            if (operand1 == null || operand2 == null || operation == null)
            {
                throw new ArgumentNullException();
            }
            if (operation.Length == 0)
            {
                throw new ArgumentException();
            }

            switch (operation)
            {
                case "+":
                    return $"{operand1} {operation} {operand2} = {operand1 + operand2}";
                case "*":
                    return $"{operand1} {operation} {operand2} = {operand1 * operand2}";
                case "/":
                    if (operand2 == 0)
                    {
                        throw new DivideByZeroException();
                    }
                    return $"{operand1} {operation} {operand2} = {operand1 / operand2}";
                default:
                    throw new ArgumentOutOfRangeException();
            }
        }
        catch (DivideByZeroException)
        {
            return "Division by zero is not allowed.";
        }
    }
}

File: java/calculator-conundrum/src/main/java/CalculatorConundrum.java
class IllegalOperationException extends RuntimeException {
  public IllegalOperationException(String message) {
    super(message);
  }

  public IllegalOperationException(String message, Throwable cause) {
    super(message, cause);
  }
}

class CalculatorConundrum {

  public String calculate(int operand1, int operand2, String operation) {
    if (operation == null) throw new IllegalArgumentException("Operation cannot be null");

    if (operation.isEmpty()) throw new IllegalArgumentException("Operation cannot be empty");

    if (!operation.equals("+") && !operation.equals("*") && !operation.equals("/"))
      throw new IllegalOperationException("Operation '" + operation + "' does not exist");

    int result;

    try {
      switch (operation) {
        case "+":
          result = operand1 + operand2;
          break;
        case "*":
          result = operand1 * operand2;
          break;
        case "/":
          result = operand1 / operand2; // can throw ArithmeticException
          break;
        default:
          throw new IllegalOperationException("Unknown operation");
      }
    } catch (ArithmeticException e) {
      throw new IllegalOperationException("Division by zero is not allowed", e);
    }

    return operand1 + " " + operation + " " + operand2 + " = " + result;
  }
}
